ashlar/reg.py

```diff
@@ -848,19 +848,28 @@ def _sanitize_channels(self, channels):
         return channels
 
     def _load_correction_profiles(self, dfp_path, ffp_path):
-        if dfp_path or ffp_path:
-            c = max(self.channels) + 1
+        if not dfp_path and not ffp_path:
+            self.do_correction = False
+        else:
+            c = self.aligner.metadata.num_channels
+            default_shape = (1, 1, c) if c in (1, 3, 4) else (c, 1, 1)
             self.dfp = np.atleast_3d(
-                skimage.io.imread(dfp_path) if dfp_path else np.zeros(c)
+                skimage.io.imread(dfp_path) if dfp_path else np.zeros(default_shape)
             )
             self.ffp = np.atleast_3d(
-                skimage.io.imread(ffp_path) if ffp_path else np.ones(c)
+                skimage.io.imread(ffp_path) if ffp_path else np.ones(default_shape)
             )
+            if (self.dfp.ndim, self.ffp.ndim) != (3, 3):
+                raise ValueError('Dimension of illumination profiles does not match target image')
+            if c in (1, 3, 4):
+                self.dfp = np.moveaxis(self.dfp, -1, 0)
```

Comment:
Also should probably use `2` instead of `-1 here, since we have a fixed number of dimensions.

Reply:
Good point! It's specified in the [next commit](https://github.com/labsyspharm/ashlar/pull/43/commits/c4960f169546c325484084ccfcd86cbf53e59935). Thanks!